ionizer/ops.py

```diff
@@ -42,6 +45,8 @@ class GPI(Operation):
     num_wires = 1
     num_params = 1
     ndim_params = (0,)
+    grad_method = "A"
+    parameter_frequencies = [(1,)]
```

Comment:
Please check into the documentation of the parameter_frequencies - it has to do with the eigenvalues of the generator, and is not a fixed value for all single-qubit operations. It's important because I believe the parameter-shift rule generator uses the frequencies to determine what the shift values and gradient recipe should be.

Reply:
Please also define the `generator` method for the operation.